src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.Unix.cs

```diff
@@ -10,7 +11,7 @@ public partial class DebugProvider
     {
         private static readonly bool s_shouldWriteToStdErr = Environment.GetEnvironmentVariable("COMPlus_DebugWriteToStdErr") == "1";
 
-        public static void FailCore(string stackTrace, string message, string detailMessage, string errorSource)
+        public static void FailCore(string stackTrace, string? message, string? detailMessage, string errorSource)
```

Comment:
It looks like `WriteCore`in this class can take a null message, no? https://github.com/dotnet/coreclr/blob/master/src/vm/debugdebugger.cpp#L259

Reply:
It looks like @stephentoub recommended to make it not accept nulls... but this is a public API right?

Reply:
It's new in 3.0, and it's not meant to be used by any code other than corefx. We can choose whatever signature we want.